spec/models/concern/document_template_concern_spec.rb

```diff
@@ -0,0 +1,251 @@
+              .update_attributes(value: 'libelle1')
+
+            dossier.champs
+              .select { |champ| champ.libelle == 'libelleB' }
+              .first
+              .update_attributes(value: 'libelle2')
+          end
+
+          it { is_expected.to eq('libelle1 libelle2') }
+        end
+      end
+    end
+
+    context 'when the dossier has a motivation' do
+      let(:dossier) { create(:dossier, motivation: 'motivation') }
+
+      context 'and the template has some dossier tags' do
+        let(:template) { '--motivation-- --numéro du dossier--' }
+
+        it { is_expected.to eq("motivation #{dossier.id}") }
+      end
+    end
+
+    context 'when the procedure has a type de champ prive named libelleA' do
+      let(:types_de_champ_private) { [create(:type_de_champ_private, libelle: 'libelleA')] }
+
+      context 'and the are used in the template' do
+        let(:template) { '--libelleA--' }
+
+        context 'and its value in the dossier are not nil' do
+          before { dossier.champs_private.first.update_attributes(value: 'libelle1') }
+
+          it { is_expected.to eq('libelle1') }
+        end
+      end
+    end
+
+    context 'when the procedure has 2 types de champ date and datetime' do
+      let(:types_de_champ) do
+        [create(:type_de_champ_public, libelle: 'date', type_champ: 'date'),
+         create(:type_de_champ_public, libelle: 'datetime', type_champ: 'datetime')]
+      end
+
+      context 'and the are used in the template' do
+        let(:template) { '--date-- --datetime--' }
+
+        context 'and its value in the dossier are not nil' do
+          before do
+            dossier.champs
+              .select { |champ| champ.type_champ == 'date' }
+              .first
+              .update_attributes(value: '2017-04-15')
+
+            dossier.champs
+              .select { |champ| champ.type_champ == 'datetime' }
+              .first
+              .update_attributes(value: '13/09/2017 09:00')
+          end
+
+          it { is_expected.to eq('15/04/2017 13/09/2017 09:00') }
+        end
+      end
+    end
+
+    context "when the template has a libellé procédure tag" do
+      let(:template) { '--libelle_procedure--' }
+
+      it { is_expected.to eq('Une magnifique procédure') }
+    end
+
+    context "when the template has a date de décision tag" do
+      let(:template) { '--date_de_decision--' }
+
+      context "and the dossier has a date de décision" do
+        let!(:dossier) { create(:dossier, processed_at: DateTime.new(2005, 3, 12), procedure: procedure, individual: individual, entreprise: entreprise) }
+
+        it { is_expected.to eq('12/03/2005') }
+      end
+
+      context "and the dossier has no date de décision" do
+        it { is_expected.to eq('') }
+      end
+    end
+
+    context "when the template has a lien_dossier tag" do
+      let(:template) { '--lien_dossier--' }
+
+      it { is_expected.to include("/users/dossiers/#{dossier.id}/recapitulatif") }
```

Comment:
rajouter le protocole + nom de domaine ?

Reply:
Peut-être en mettant un bouchon sur la génération du lien ?

Reply:
Non mais il doit déjà le faire j'imagine en prenant la valeur de config dans `test.rb`:
`config.action_mailer.default_url_options = { :host => 'localhost:3000' }`

Tu dois pouvoir écrire ton test :
`it { is_expected.to include("http://localhost:3000/users/dossiers/#{dossier.id}/recapitulatif") }`